views/default/plugins/event_calendar/settings.php

```diff
@@ -1,5 +1,17 @@
 <?php
 
+if (event_calendar_is_upgrade_available()) {
+	echo '<div class="elgg-admin-notices">';
+	echo '<p>';
+	echo elgg_view('output/url', array(
+		'text' => elgg_echo('event_calendar:upgrade'),
+		'href' => 'action/event_calendar/upgrade',
+		'is_action' => true,
+	));
+	echo '</p>';
+	echo '</div>';
+}
```

Comment:
Is there any way to execute the upgrade action directly without user input in the callback function within the 'upgrade', 'system' hook? The admin notice would only get shown after the admin executes the core upgrade, so the plugin hook callback would not result in the admin notice to show automatically if an upgrade for event_calendar is available. And if the core update is executed anyway it might be an opportunity to update event_calendar (silently), too. On the other hand, such a silent upgrade of a plugin might not be transparent and I wouldn't want to hide what's going on from the site admin. Also, the plugin specific upgrade might take a while (as in case with the converting of the annotations to relationships for the event_calendar plugin some sites might have many of them to process).